docs/source/en/model_doc/depth_anything.md

```diff
@@ -14,7 +14,7 @@ rendered properly in your Markdown viewer.
 
 -->
 
-# Depth Anything
+# Depth Anything and Depth Anything V2
```

Comment:
I'm in favor of not polluting this docs and instead add a new docs just for v2, as it's also a new paper: https://arxiv.org/abs/2406.09414.

This can be done in a similar way to how we did it for Flan-T5 compared to the original T5: https://github.com/huggingface/transformers/blob/main/docs/source/en/model_doc/flan-t5.md

Reply:
Agreed here - I'm happy for updates to the script if it's just a few lines so we can convert the checkpoints, but if adding the model into the library it should have its own model page

Reply:
I wasn't sure how to deal with this. There are no modelling changes, the conversion script is inside the same directory as the previous checkpoints, and I felt it was weird to have a documentation page about a new model that actually refers to the same implementation as before. In my opinion, it's clearer to mention both in the same page so readers understand it's the same model architecture. We can use a single name in the title if that's preferred, and maybe improve the description in the body of the page making sure we mention both papers.

Happy to work on another solution if there's consensus. These are the options I see:
1. Remove the doc updates, as in the original version of this PR that was approved.
2. Create a new documentation page for Depth Anything V2. It'd be essentially a duplicate of the Depth Anything page, except the paper would be updated and the snippets would use the new model ids.
3. Use the same page for both, as in the current version of this PR, maybe tweaking as needed.

Reply:
I'm in favor of option 2 since we did the same for other models in the past